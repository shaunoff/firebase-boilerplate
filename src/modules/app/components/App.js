import React from 'react';

const App = ({auth, signIn, signOut }) => {
  if(auth.status === "AWAITING_AUTH_RESPONSE"){
    return <h2>Loading...</h2>
  }
  return (
      <div>
				<h2>{`Status: ${auth.status === "SIGNED_IN" ? "logged in" : "logged out"}`}</h2>
				{auth.status === "SIGNED_IN" ? <button onClick={signOut}>Sign Out</button> : <button onClick={signIn}>Sign In</button>}
      </div>
  );
};

export default App


// { auth.status === 'ANONYMOUS' && <SignIn signIn={signIn}/> }
// { auth.status === 'SIGNED_IN' && <CurrentUser auth={auth} signOut={signOut} />}
// { auth.status === 'AWAITING_AUTH_RESPONSE' && <Loading /> }
