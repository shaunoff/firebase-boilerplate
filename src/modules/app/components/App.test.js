import React from 'react';
import {shallow} from 'enzyme'

import App from './App'

const loggedOut = {
	status: 'ANONYMOUS'
}
const loggedIn = {
	status: 'SIGNED_IN'
}
const loading = {
	status: 'AWAITING_AUTH_RESPONSE'
}

const appLoggedIn = shallow(<App auth={loggedIn}/>);
const appLoggedOut = shallow(<App auth={loggedOut}/>);
const appLoading = shallow(<App auth={loading}/>);

describe('App Component', ()=>{
	it('Renders Loading State correctly', ()=>{
		expect(appLoading.find('h2').text()).toEqual('Loading...')
	})
	it('Renders Logged Out Status correctly', ()=>{
		expect(appLoggedOut.find('h2').text()).toEqual('Status: logged out')
		expect(appLoggedOut.find('button').text()).toEqual('Sign In')
	})
	it('Renders Logged In Status correctly', ()=>{
		expect(appLoggedIn.find('h2').text()).toEqual('Status: logged in')
		expect(appLoggedIn.find('button').text()).toEqual('Sign Out')
	})
	it('Renders Logged In App', ()=>{
		expect(appLoggedIn).toMatchSnapshot()
	})
	it('Renders Logged Out App', ()=>{
		expect(appLoggedOut).toMatchSnapshot()
	})
})
