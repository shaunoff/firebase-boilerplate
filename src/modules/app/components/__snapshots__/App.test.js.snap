// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component Renders Logged In App 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    auth={
        Object {
            "status": "SIGNED_IN",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Status: logged in
</h2>,
        <button
          onClick={undefined}
>
          Sign Out
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Status: logged in",
        },
        "ref": null,
        "rendered": "Status: logged in",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign Out",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Sign Out",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Status: logged in
</h2>,
          <button
            onClick={undefined}
>
            Sign Out
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Status: logged in",
          },
          "ref": null,
          "rendered": "Status: logged in",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign Out",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Sign Out",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;

exports[`App Component Renders Logged Out App 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <App
    auth={
        Object {
            "status": "ANONYMOUS",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h2>
          Status: logged out
</h2>,
        <button
          onClick={undefined}
>
          Sign In
</button>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Status: logged out",
        },
        "ref": null,
        "rendered": "Status: logged out",
        "type": "h2",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Sign In",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Sign In",
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h2>
            Status: logged out
</h2>,
          <button
            onClick={undefined}
>
            Sign In
</button>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Status: logged out",
          },
          "ref": null,
          "rendered": "Status: logged out",
          "type": "h2",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Sign In",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Sign In",
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
